hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -858,15 +860,29 @@ private static int parsePartNumberMarker(String partNumberMarker) {
     return partMarker;
   }
 
-  private static long parseOzoneDate(String ozoneDateStr) throws OS3Exception {
+  // Parses date string and return long representation. Returns an
+  // empty if DateStr is null or invalid. Dates in the future are
+  // considered invalid.
+  private static OptionalLong parseAndValidateDate(String ozoneDateStr)
+      throws OS3Exception {
```

Comment:
No longer thrown:

```suggestion
  private static OptionalLong parseAndValidateDate(String ozoneDateStr) {
```

(Also in `checkCopySourceModificationTime`.)